src/Fixer/FunctionNotation/UseArrowFunctionsFixer.php

```diff
@@ -0,0 +1,214 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\FunctionNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Gregor Harlan
+ */
+final class UseArrowFunctionsFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Anonymous functions with one-liner return statement must use arrow functions.',
+            [
+                new VersionSpecificCodeSample(
+                    <<<'SAMPLE'
+<?php
+foo(function ($a) use ($b) {
+    return $a + $b;
+});
+
+SAMPLE
+                    ,
+                    new VersionSpecification(70400)
+                ),
+            ],
+            null,
+            'Risky when using "isset()" on outside variables that are not imported with "use ()".'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70400 && $tokens->isAllTokenKindsFound([T_FUNCTION, T_RETURN]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $analyzer = new TokensAnalyzer($tokens);
+
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION) || !$analyzer->isLambda($index)) {
+                continue;
+            }
+
+            // Find parameters end
+            // Abort if they are multinlined
+
+            $parametersStart = $tokens->getNextMeaningfulToken($index);
+
+            if ($tokens[$parametersStart]->isGivenKind(CT::T_RETURN_REF)) {
+                $parametersStart = $tokens->getNextMeaningfulToken($parametersStart);
+            }
+
+            $parametersEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $parametersStart);
+
+            if ($this->isMultilined($tokens, $parametersStart, $parametersEnd)) {
+                continue;
+            }
+
+            // Find `use ()` start and end
+            // Abort if it contains reference variables
+
+            $next = $tokens->getNextMeaningfulToken($parametersEnd);
+
+            $useStart = null;
+            $useEnd = null;
+
+            if ($tokens[$next]->isGivenKind(CT::T_USE_LAMBDA)) {
+                $useStart = $next;
+
+                if ($tokens[$useStart - 1]->isGivenKind(T_WHITESPACE)) {
+                    --$useStart;
```

Comment:
missing coverage from when this isn't the case, like a minimal case with as little spacing as possible; `return function()use($a) {`

Reply:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4778/commits/30ff1ed148313730c8cefb9858e89ab5b9cdc848